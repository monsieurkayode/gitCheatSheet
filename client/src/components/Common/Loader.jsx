import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import PreloaderIcon from 'react-preloader-icon';
import Oval from 'react-preloader-icon/loaders/Oval';

const LoaderWrapper = styled.div`
  &.loader {
    flex: ${({ loading }) => (loading ? 1 : 0)}
  }
`;
/**
 *
 * @function Loader
 *
 * @param {object} props
 *
 * @return {JSX} JSX
 */
const Loader = ({
  size,
  strokeColor,
  inline,
  loading
}) => (
  <LoaderWrapper
    style={{
      visibility: loading ? 'visible' : 'hidden'
    }}
    loading={loading}
    className={`${inline ? 'btn-loader' : 'loader'}`}
  >
    <PreloaderIcon
      loader={Oval}
      size={size}
      strokeWidth={12}
      strokeColor={strokeColor}
      duration={1200}
    />
  </LoaderWrapper>
);

Loader.defaultProps = {
  inline: false
};

Loader.propTypes = {
  size: PropTypes.number.isRequired,
  strokeColor: PropTypes.string.isRequired,
  inline: PropTypes.bool,
  loading: PropTypes.bool.isRequired
};

export default Loader;
