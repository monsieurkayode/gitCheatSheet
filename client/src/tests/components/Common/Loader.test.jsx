import React from 'react';
import { mount } from 'enzyme';
import Loader from '../../../components/Common/Loader';

const setup = () => {
  const props = {
    size: 20,
    strokeColor: 'white',
    loading: true
  };

  return mount(<Loader {...props} />);
};

describe('Loader Component', () => {
  const wrapper = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
