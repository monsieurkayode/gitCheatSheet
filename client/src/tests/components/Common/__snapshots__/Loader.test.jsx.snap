// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should render correctly 1`] = `
<Loader
  inline={false}
  loading={true}
  size={20}
  strokeColor="white"
>
  <styled.div
    className="loader"
    loading={true}
    style={
      Object {
        "visibility": "visible",
      }
    }
  >
    <StyledComponent
      className="loader"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "bJkqPH",
            "rules": Array [
              "
  &.loader {
    flex: ",
              [Function],
              "
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      loading={true}
      style={
        Object {
          "visibility": "visible",
        }
      }
    >
      <div
        className="loader sc-bdVaJa bJkqPH"
        style={
          Object {
            "visibility": "visible",
          }
        }
      >
        <PreloaderIcon
          className=""
          duration={1200}
          loader={[Function]}
          size={20}
          strokeColor="white"
          strokeWidth={12}
          style={Object {}}
          unit="px"
        >
          <div
            className="preloader-icon "
            style={
              Object {
                "height": "20px",
                "width": "20px",
              }
            }
          >
            <div
              className="preloader-icon__inner"
              style={
                Object {
                  "height": "100%",
                  "lineHeight": 0,
                  "overflow": "hidden",
                  "position": "relative",
                  "transform": "translate3d(0, 0, 0)",
                  "width": "100%",
                }
              }
            >
              <em
                className="preloader-icon__title"
                style={
                  Object {
                    "fontSize": 0,
                    "height": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "width": 0,
                  }
                }
              >
                Loading...
              </em>
              <Oval
                duration={1200}
                strokeColor="white"
                strokeWidth={12}
              >
                <div
                  className="preloader-icon__oval"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  <svg
                    height="100%"
                    stroke="white"
                    viewBox="0 0 102 102"
                    width="100%"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                    >
                      <g
                        strokeWidth={12}
                        transform="translate(1 1)"
                      >
                        <circle
                          cx="50"
                          cy="50"
                          r={44}
                          stroke="white"
                          strokeOpacity=".5"
                        />
                        <path
                          d="M50,6 A44,44,0,0,1,94,50"
                        />
                      </g>
                    </g>
                  </svg>
                </div>
              </Oval>
            </div>
          </div>
        </PreloaderIcon>
      </div>
    </StyledComponent>
  </styled.div>
</Loader>
`;
